tests/splines/matrix.cpp

```diff
@@ -339,6 +339,35 @@ TEST_P(MatrixSizesFixture, 2x2Blocks)
     solve_and_validate(*matrix);
 }
 
+TEST_P(MatrixSizesFixture, PeriodicBand)
+{
+    auto const [N, k] = GetParam();
+
+    // Build a positive-definite symmetric full-rank band matrix permuted in such a way the band is shifted
+    for (std::ptrdiff_t s(-k); s < (std::ptrdiff_t)k + 1; ++s) {
+        std::unique_ptr<ddc::detail::SplinesLinearProblem<Kokkos::DefaultExecutionSpace>> matrix
+                = ddc::detail::SplinesLinearProblemMaker::make_new_periodic_band_matrix<
+                        Kokkos::DefaultExecutionSpace>(
+                        N,
+                        (std::ptrdiff_t)k - s,
+                        (std::ptrdiff_t)k + s,
+                        false);
+        for (std::size_t i(0); i < N; ++i) {
+            for (std::size_t j(0); j < N; ++j) {
+                std::ptrdiff_t diag
+                        = ddc::detail::modulo((std::ptrdiff_t)(j - i), (std::ptrdiff_t)N);
+                if (diag == s || diag == N + s) {
+                    matrix->set_element(i, j, .5);
+                } else if (diag <= s + k || diag >= N + s - k) {
+                    matrix->set_element(i, j, -1. / k);
+                }
```

Comment:
What guarantee do we have on the invertibility of these matrices ?

Reply:
> I think the value of `s` has no impact on invertibility because this is just row interchanges. I suspect the "problem" to come from diagonal-dominancy.

What do you mean ? Does any underlying solver require such assumption ?

Reply:
I forgot to answer this. `s` shifts the diagonal so I think this is just row-interchanges and thus has no impact on invertibility. But it impacts storage and LAPACK calls, of course. Hope I am right about this statement.

Reply:
Thinking about it again, I think the invertibility of the whole matrix is guaranteed but not the one of the top-left or bottom-right corners, which have their own solvers.